sktime/datatypes/_panel/_check.py

```diff
@@ -171,41 +173,56 @@ def check_pdmultiindex_panel(obj, return_metadata=False, var_name="obj"):
         msg = f"{var_name} must have a MultiIndex with 2 levels, found {nlevels}"
         return _ret(False, msg, None, return_metadata)
 
+    # check that no dtype is object
+    if "object" in obj.dtypes.values:
+        msg = f"{var_name} should not have column of 'object' dtype"
+        return _ret(False, msg, None, return_metadata)
+
+    # check whether the time index is of valid type
+    if not is_in_valid_index_types(obj.index.get_level_values(-1)):
+        msg = (
+            f"{type(obj.index)} is not supported for {var_name}, use "
+            f"one of {VALID_INDEX_TYPES} or integer index instead."
+        )
+        return _ret(False, msg, None, return_metadata)
+
+    # inst_inds_names = obj.index.names[0:-1]
+    # time_inds_names = obj.index.names[-1]
+    time_obj = obj.reset_index(-1).drop(obj.columns, axis=1)
+    time_grp = time_obj.groupby(level=0, group_keys=True, as_index=True)
+    inst_inds = time_obj.index.unique()
+
     # check instance index being integer or range index
-    instind = obj.index.get_level_values(0)
-    if not is_in_valid_multiindex_types(instind):
+    if not is_in_valid_multiindex_types(inst_inds):
         msg = (
             f"instance index (first/highest index) must be {VALID_MULTIINDEX_TYPES}, "
-            f"integer index, but found {type(instind)}"
+            f"integer index, but found {type(inst_inds)}"
         )
         return _ret(False, msg, None, return_metadata)
 
-    inst_inds = obj.index.get_level_values(0).unique()
-    # inst_inds = np.unique(obj.index.get_level_values(0))
+    is_equally_spaced = (
+        time_grp.diff().groupby(level=0, group_keys=True, as_index=True).nunique()
+    )
 
-    check_res = [
-        check_pddataframe_series(obj.loc[i], return_metadata=True) for i in inst_inds
-    ]
-    bad_inds = [i for i in range(len(inst_inds)) if not check_res[i][0]]
+    unique_diff = is_equally_spaced.iloc[:, 0].unique()
+    is_equal_length = time_grp.count()
 
-    if len(bad_inds) > 0:
+    # Check time index is ordered in time
+    if not all(unique_diff >= 0):
         msg = (
-            f"{var_name}.loc[i] must be Series of mtype pd.DataFrame,"
-            f" not at i={bad_inds}"
+            f"The (time) index of {var_name} must be sorted monotonically increasing, "
+            f"but found: {obj.index.get_level_values(-1)}"
         )
         return _ret(False, msg, None, return_metadata)
 
     metadata = dict()
-    metadata["is_univariate"] = np.all([res[2]["is_univariate"] for res in check_res])
-    metadata["is_equally_spaced"] = np.all(
-        [res[2]["is_equally_spaced"] for res in check_res]
-    )
-    metadata["is_empty"] = np.any([res[2]["is_empty"] for res in check_res])
+    metadata["is_univariate"] = len(obj.columns) < 2
+    metadata["is_equally_spaced"] = unique_diff.shape[0] == 1
+    metadata["is_empty"] = len(obj.index) < 1 or len(obj.columns) < 1
```

Comment:
Apologies if this is a stupid question. My understanding is that  1) `"is_empty"` checks if one or more of the series in the panel are empty and 2) `obj` is our multi-index dataframe representing our panel data. I don't understand how line 221 checks whether there is at least one instance if `obj` is our multi-index dataframe? (perhaps it isn't a multi-index dataframe and I should walkthrough the code in debug mode and see what `obj` actually is here).

Reply:
@KishManani, to my understanding, this checks whether *all* of the series are empty, so should be fine

Reply:
(it cannot happen that just one of the instances is empty, since in a `pd.DataFrame` then the instance just wouldn´t be there, no row index)

Reply:
Ah! I was confused by the docstring at the top of the module:

> "is_empty": bool, True iff one or more of the series in the panel are empty

Perhaps we should change this to `"... True iff all of the series in the panel are empty"`.

Reply:
Hi this is actually carried over from the check in check_pddataframe_series. Previously, the check was made for every instance, which does not make sense since in a multiindex dataframe, it is not possible that one instance has an index / column and the other one not.